src/imap/mod.rs

```diff
@@ -1501,6 +1504,18 @@ pub(crate) async fn prefetch_should_download(
     headers: &[mailparse::MailHeader<'_>],
     show_emails: ShowEmails,
 ) -> Result<bool> {
+    if let Some(rfc724_mid) = headers.get_header_value(HeaderDef::MessageId) {
+        if let Some(group_id) = dc_extract_grpid_from_rfc724_mid(&rfc724_mid) {
+            if let Ok((chat_id, _, _)) = get_chat_id_by_grpid(context, group_id).await {
+                if !chat_id.is_unset() {
```

Comment:
Why check for existing chat_id? It will not find any chat if blocked user adds you to a group you don't know about yet, but all users there will think you are in the group now.

Reply:
The thing is, I don't want blocked users to be able to add me to a group. They could just circumvent my block by adding me to a group, which would make the block useless.

I checked that if there is an unblocked member in the group, and they send a message to the group, then a group will be created. So, the inconsistent group state will be only for a short time.

Reply:
OK, if it only works when non-blocked user sends a message, then it makes sense.